dashboard/src/actions/comparisonActions.js

```diff
@@ -139,32 +184,37 @@ export const parseChartData = () => (dispatch, getState) => {
   });
 };
 
-export const toggleCompareSwitch = () => (dispatch, getState) => {
-  dispatch({
-    type: TYPES.TOGGLE_COMPARE_SWITCH,
-    payload: !getState().comparison.isCompareSwitchChecked,
-  });
-};
+export const toggleCompareSwitch = () => ({
+  type: TYPES.TOGGLE_COMPARE_SWITCH,
+});
 
 export const setSelectedId = (isChecked, rId) => (dispatch, getState) => {
-  let selectedIds = [...getState().comparison.selectedResourceIds];
-  if (isChecked) {
-    selectedIds = [...selectedIds, rId];
+  const prev = [...getState().comparison.selectedResourceIds];
+  const selectedIds = isChecked
+    ? [...prev, rId]
+    : prev.filter((id) => id !== rId);
```

Comment:
This is slightly disappointing:  we create a copy of the `selectedResourceIds` at line 192 and then we immediately make a copy of that at either line 194 or 195.  How about just creating a reference to the state?
```javascript
  const prev = getState().comparison.selectedResourceIds;
  const selectedIds = isChecked
    ? [...prev, rId]
    : prev.filter((id) => id !== rId);
```